### PR TITLE
OpenSSL: add support for watchos

### PR DESCRIPTION
**Pull request by paulo-coutinho**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openssl/1.x.x/conanfile.py [hunk 1]
```diff
@@ -463,8 +463,8 @@ def _configure_args(self):
         if self._full_version >= "1.1.0":
             args.append("--debug" if self.settings.build_type == "Debug" else "--release")
 
-        if self.settings.os == "tvOS":
-            args.append(" -DNO_FORK") # fork is not available on tvOS
+        if self.settings.os == "tvOS" or self.settings.os == "watchOS":
```
Comment (SpaceIm) [inline recipes/openssl/1.x.x/conanfile.py:466]:
```suggestion
        if self.settings.os in ["tvOS", "watchOS"]:
```

File: recipes/openssl/1.x.x/conanfile.py [hunk 2]
```diff
@@ -689,8 +689,8 @@ def build(self):
                 env_vars["CROSS_TOP"] = os.path.dirname(os.path.dirname(xcrun.sdk_path))
             with tools.environment_append(env_vars):
                 if self._full_version >= "1.1.0":
-                    if self.settings.os == "tvOS":
-                        tools.patch(patch_file=os.path.join("patches", "1.1.1-tvos.patch"),
+                    if self.settings.os == "tvOS" or self.settings.os == "watchOS":
```
Comment (SpaceIm) [inline recipes/openssl/1.x.x/conanfile.py:692]:
```suggestion
                    if self.settings.os in ["tvOS", "watchOS"]:
```

**Comment by SpaceIm** [comment]
Update the title please to show which recipe is modified.

**Comment by paulo-coutinho** [comment]
What is wrong?

```
[HOOK - conan-center.py] pre_build(): ERROR: [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not managed correctly. Please remove it for shared option: del self.options.fpic (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H007) 
```

**Comment by jgsogo** [comment]
Hi! We merged a more restrictive hook in the previous days: https://github.com/conan-io/hooks/pull/286

I know this is not related to the changes in your PR, but please, fix the error following the documented KB, just add the following to the `configure()` method to remove `fPIC` option when building a shared library.

```
    def configure(self):
        ...
        if self.options.shared:
            del self.options.fPIC
```

Thanks!

**Comment by uilianries** [comment]
@paulo-coutinho https://github.com/paulo-coutinho/conan-center-index/pull/1

**Comment by paulo-coutinho** [comment]
Hi,

The changes was done.

Thanks.

**Comment by paulo-coutinho** [comment]
Fixed place "configure" instead of  "config_options"

**Comment by SpaceIm** [comment]
you have created another configure() method instead of adding logic in already existing one.

**Comment by paulo-coutinho** [comment]
Yes, sorry. Fixed now.

**Comment by SpaceIm** [comment]
Replace all the occurrences of `if self.options.fPIC` by `if self.options.get_safe("fPIC", True)`, and it should work.

**Comment by paulo-coutinho** [comment]
Hi,

Fixed.

Thanks.

**Review by madebr — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by paulo-coutinho** [comment]
I get error when compile:
https://github.com/conan-io/conan-center-index/issues/5061

**Pull request merged**